src/File.php

```diff
@@ -485,6 +489,65 @@ public function canDelete($member = null)
             ->canDelete($this->ID, $member);
     }
 
+
+    /**
+     * Check if the File has a CanViewType set or inherited at "LoggedInUsers or above"
+     *
+     * @return bool
+     */
+    public function hasRestrictedAccess(): bool
+    {
+        return $this->hasRestrictedPermissions($this);
+    }
+
+    /**
+     * Recursively determine whether a File has, or inherits, restricted permissions.
+     *
+     * This custom permission check is essentially a couldView() if the record was published for an
+     * anonymous (not-logged-in) user
+     *
+     * Ideally we wouldn't need this custom logic and instead use File->canView() which utilises
+     * InheritedPermission::batchPermissionCheck() and we'd wrap this in `Member::actAs(null,` and
+     * `Versioned::set_reading_mode('Stage.Stage')`, however the anonymous user will fail the
+     * `Versioned::canViewVersioned()` check
```

Comment:
> I think you could say here that you want a canView check but without other modules involved.

It's not actually a canView() check since that involves a `$member`.  This is a `$memberless` permission check, which is something I don't think we do anywhere else.  The docblock explanation is so people understand why we didn't just use existing methods

> did you consider dogfooding this new API?

By this do you mean updating the existing File::canView() to use this instead of the  InheritedPermissions logic it uses now?

Reply:
If this logic is new then all good, was just suggesting we reuse this new API if possible

Reply:
I've updated the comment to spell it out clearer that it's a member-less permission check